File: sample-conference-videochat/sample-conference-videochat/Components/ChatScreen/Views/ChatCollectionView/ChatCallIncomingCell/ChatCallIncomingCell.m
//
//  ChatCallIncomingCell.m
//  sample-conference-videochat
//
//  Created by Injoit on 25.03.2021.
//  Copyright © 2021 Quickblox. All rights reserved.
//

#import "ChatCallIncomingCell.h"
#import "UIView+Chat.h"
#import "CALayer+Chat.h"

@implementation ChatCallIncomingCell

- (void)awakeFromNib {
    [super awakeFromNib];
    
    [self.joinButton setRoundViewWithCornerRadius:15.0f];
    self.bubbleImageView.backgroundColor = UIColor.whiteColor;
    UIColor *shadowColor = [UIColor colorWithRed:0.85f green:0.9f blue:1.0f alpha:1.0f];
    [self.layer applyShadowWithColor:shadowColor
                               alpha:1.0f
                                forX:0.0f
                                forY:3.0f
                                blur:48.0f
                              spread:0.0f
                                path:nil];
}

+ (ChatCellLayoutModel)layoutModel {
    
    ChatCellLayoutModel defaultLayoutModel = [super layoutModel];
    defaultLayoutModel.containerInsets = UIEdgeInsetsMake(0, 24, 12, 16);
    defaultLayoutModel.avatarSize = CGSizeMake(40, 40);
    return defaultLayoutModel;
}

- (void)layoutSubviews {
    CGFloat cornerRadius = 20.0f;
    CAShapeLayer *layer = CAShapeLayer.new;
    layer.frame = self.bubbleImageView.layer.bounds;
    UIRectCorner roundingCorners = UIRectCornerTopLeft | UIRectCornerTopRight | UIRectCornerBottomRight;
    UIBezierPath *bPath = [UIBezierPath bezierPathWithRoundedRect:self.bubbleImageView.bounds
                                                byRoundingCorners:roundingCorners
                                                      cornerRadii:CGSizeMake(cornerRadius, cornerRadius)];
    layer.path = bPath.CGPath;
    self.bubbleImageView.layer.mask = layer;
}

- (IBAction)didTapJoinButton:(UIButton *)sender {
    if (self.didPressJoinButton) {
        self.didPressJoinButton();
    }
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/Components/ChatScreen/Chat/CollectionView/CollectionCells/ChatCallStatusCell/ChatCallIncomingCell.swift
//
//  ChatCallIncomingCell.swift
//  sample-conference-videochat-swift
//
//  Created by Injoit on 5/8/20.
//  Copyright © 2020 QuickBlox. All rights reserved.
//

import UIKit
import TTTAttributedLabel

class ChatCallIncomingCell: ChatCell {
    
    @IBOutlet weak var streamLabel: TTTAttributedLabel!
    @IBOutlet weak var joinButton: UIButton!
    
    /**
     *  Join user to conferense block action.
     */
    var didPressJoinButton: (() -> Void)?
    
    override func draw(_ rect: CGRect) {
        super.draw(rect)
        
        let roundingCorners:UIRectCorner = [.topLeft, .topRight, .bottomRight]
        let layer = CAShapeLayer()
        layer.frame = bubbleImageView.layer.bounds
        let bPath = UIBezierPath(roundedRect: bubbleImageView.bounds,
                                 byRoundingCorners: roundingCorners,
                                 cornerRadii: CGSize(width: 20, height: 20))
        layer.path = bPath.cgPath
        bubbleImageView.layer.mask = layer
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()

        joinButton.setRoundView(cornerRadius: 15.0)
        bubbleImageView.backgroundColor = .white
        layer.applyShadow(color: #colorLiteral(red: 0.8452011943, green: 0.8963350058, blue: 1, alpha: 1), alpha: 1.0, y: 3.0, blur: 48.0)
    }
    
    @IBAction func didTapJoinButton(_ sender: UIButton) {
        didPressJoinButton?()
    }
    
    override class func layoutModel() -> ChatCellLayoutModel {
        var defaultLayoutModel = super.layoutModel()
        let containerInsets = UIEdgeInsets(top: 0.0, left: 24.0, bottom: 12.0, right: 16.0)
        defaultLayoutModel.containerInsets = containerInsets
        defaultLayoutModel.avatarSize = CGSize(width: 40.0, height: 40.0)
        
        return defaultLayoutModel
    }
}
